torch/cuda/streams.py

```diff
@@ -17,7 +17,7 @@ class Stream(torch._C._CudaStreamBase):
                                  represent higher priorities.
     """
 
-    def __new__(cls, device=None, priority=0, **kwargs):
+    def __new__(cls, device=None, priority=0 if torch.version.cuda else 1, **kwargs):
```

Comment:
This is seems bad. Why do you have to do this?

Reply:
Following CUDA semantics, priority 0 is default.  There isn't any way to distinguish between a caller not specifying a priority (defaulting to 0 i.e. default priority), and a ROCm user having specifically asked for 0==high priority after a call to `priority_range()`.

Reply:
OK. Then my broader question is, how come the code for ROCm has to be different than the code for CUDA? Are ROCm and CUDA different, or is the claim that the CUDA behavior is bad and the ROCm version of this behavior is better?